PosthogFC.py

```diff
@@ -136,25 +136,152 @@ def posthog_system_info():
+                    overlay_preferences["height"] = overlay_group.GetInt("Height", 0)
+
+                return overlay_preferences
+
+        return {}
+
+    preferences = {
+        "enabled": TrackedPreference.bool(path="BaseApp/MainWindow/DockWindows", default=False),
+        **get_overlay_preferences(),
+    }
+
+    return {f"{prefix}_{name}": preference for name, preference in preferences.items()}
+
+
 def posthog_preferences():
     """Send some basic FreeCAD preferences to Posthog"""
     global posthog
-    general = FreeCAD.ParamGet("User parameter:BaseApp/Preferences/General")
-    main_window = FreeCAD.ParamGet("User parameter:BaseApp/Preferences/MainWindow")
-    units = FreeCAD.ParamGet("User parameter:BaseApp/Preferences/Units")
 
-    posthog.capture(
-        posthog_id,
-        event="freecad_preferences",
-        properties={
-            "language": general.GetString("Language", FreeCADGui.getLocale()),
-            "theme": main_window.GetString("Theme", "unset"),
-            "stylesheet": main_window.GetString("Stylesheet", "unset"),
-            "geometry": main_window.GetString("Geometry", "unset"),
-            "overlay_stylesheet": main_window.GetString("OverlayActiveStyleSheet", "unset"),
-            "default_unit_schema": units.GetInt("UserSchema", 0),
-        },
-    )
+    preferences = {
+        "language": TrackedPreference.string(
+            path="BaseApp/Preferences/General/Language",
+            default=FreeCADGui.getLocale(),
+        ),
+        "theme": TrackedPreference.string(
+            path="BaseApp/Preferences/MainWindow/Theme", default="unset"
+        ),
+        "stylesheet": TrackedPreference.string(
+            path="BaseApp/Preferences/MainWindow/StyleSheet",
+            default="unset",
+        ),
+        "overlay_stylesheet": TrackedPreference.string(
+            path="BaseApp/Preferences/MainWindow/OverlayActiveStyleSheet",
+            default="unset",
+        ),
+        "geometry": TrackedPreference.string(
+            path="BaseApp/Preferences/MainWindow/Geometry",
+            default="unset",
+        ),
+        "overlay_stylesheet": TrackedPreference.string(
+            path="BaseApp/Preferences/MainWindow/OverlayActiveStyleSheet",
+            default="unset",
+        ),
+        "default_unit_schema": TrackedPreference.int(
+            path="BaseApp/Preferences/Units/UserSchema",
+            transform=map_enum(
+                {
+                    0: "Standard (mm, kg, s, °)",
+                    1: "MKS (m, kg, s, °)",
+                    2: "US customary (in, lb)",
+                    3: "Imperial decimal (in, lb)",
+                    4: "Building Euro (cm, m², m³)",
+                    5: "Building US (ft-in, sqft, cft)",
+                    6: "Metric small parts & CNC (mm, mm/min)",
+                    7: "Imperial for Civil Eng (ft, ft/s)",
+                    8: "FEM (mm, N, s)",
+                    9: "Meter decimal (m, m², m³)",
```

Comment:
It seems like adding the transform (here and elsewhere) is making the Addon more susceptible to changes in FreeCAD: if something changes there, like we add a new unit, we'll have to push an update. Instead, I'd suggest just transmitting the raw integer value, and do the transformation on the Dashboards (that's what I'm doing for the unit schema, for example).

Reply:
Good point, I'll remove the `map_enum`s and the function itself but I'll leave the transform param in case of it is needed for other cases like scrubbing some data, or transforming it in any other way.